### ISSUE TITLE
Improve the UX for step title in activity creation form

### ISSUE THREAD

**Issue by srish** [unstructuredstudio/zubhub#896]
Currently, only users with staff privileges can see the activity creation form. So, I am sharing screenshots below. **Note:** Please don't request staff access; it is currently not possible.

In the "Step" section in the activity creation form, it is not clear that "Step 1: Enter title here" is clickable and accepts text input. This design needs improvement. Perhaps, we can add a new input field similar to the step description for the title.

<img src=
"https://github.com/unstructuredstudio/zubhub/assets/288353/4144c4b5-6681-43b2-938d-84da6846c014" width="400px">

**Comment by Dumzy006** [comment]
Hello @srish made a design improvement to solve the step title issue
In my screenshot below, I provided an input field for text title with a place holder in it and I also added an input label  for step description just to maintain a consistent look . What do you think?


![image](https://github.com/unstructuredstudio/zubhub/assets/136137889/46b07cac-05bc-48ed-80de-ae63f2d326d8)

**Comment by Temabo** [comment]
Hi @srish 
I added an input field for step titles in the "Step 1" section of the activity creation form. Labeled it as "Step Title" with a placeholder text "Enter the step title" for clarity and ease of use. I also modified the placeholder for the Task Description to prompt the Staff to explain the task in detail.

![Activity creation form (1)](https://github.com/unstructuredstudio/zubhub/assets/105416140/2de673d1-2ac5-4ac9-85cb-cf5a97a7ab2c)

**Comment by mehreeee** [comment]
![Title](https://github.com/unstructuredstudio/zubhub/assets/105458062/14b17f63-8ba2-414a-ba59-b220acdadcc2)

@srish What do you think? I have broken down the steps so the users know they are supposed to fill out every forms

**Comment by mehreeee** [comment]
![Title](https://github.com/unstructuredstudio/zubhub/assets/105458062/60b4652c-7018-4cd0-a2a7-9d0e0dc68d3a)
![Title](https://github.com/unstructuredstudio/zubhub/assets/105458062/2536845b-39f0-4e3c-8140-af4166d2b047)

Hi, @srish I added another field with a *required* highlighted text so users can understand better. I couldn't decide which one looked better so here's both.

**Comment by ozoilo** [comment]
@srish I would like to work on this issue.

**Comment by JulietMadu** [comment]
I don't know why I'm unable to see the images from my end

**Comment by Mbeute23** [comment]
I have added an input field for Title with a placeholder enter title and a placeholder for description with enter description text here for the user to know they have to field in the require information.

![Improve the UX for step title in activity creation form #896](https://github.com/unstructuredstudio/zubhub/assets/146943654/f74389aa-8259-4e4f-bc04-a82ec3d835ed)

@srish what do you think about this design

**Comment by amemenikejudith** [comment]
Hello @srish, I have made a design improvement to solve the step title issue
In my screenshot below, I provided an input field for the title with a placeholder and added an input field for the description to maintain a user-friendly look. Let me know what you think.
![Frame 893](https://github.com/unstructuredstudio/zubhub/assets/105314523/9d0cb29d-fb0b-438a-af17-07cf948a781a)

**Comment by JulietMadu** [comment]
i added a field for the users to input step title, 
in the description field, i added icons for them to increase, bold, italic, underline align, highlight or change font style of the text
i also added a file with a title "attach files" for the users to upload the supported files of their choice @srish 
![step file](https://github.com/unstructuredstudio/zubhub/assets/128796937/36618249-57bd-4e00-a6db-c604397e8712)

**Comment by Marhiposa** [comment]
Hi @srish. [Improve the UX for step title in activity creation form](https://github.com/unstructuredstudio/zubhub/issues/896#top)
Made an improvement on the design. added a new input field similar to the step description for the title.

![Step title (1)](https://github.com/unstructuredstudio/zubhub/assets/146780247/2a3f6cb9-b245-4685-9c17-b5938cdf9194)

**Comment by Mitchell377** [comment]
so i  used auto layout for consistent spacing, i made sure the :step: and :descibed: text were bold , i also used a place holder, focus place holder which drew more attention to the text and then a character limit that will help the user narrow down his or her points as to the description of the steps.

![main group step](https://github.com/unstructuredstudio/zubhub/assets/102149002/6df3ac06-df18-4530-9250-cd9e02be1a98)

@srish  will love to hear your feedback

**Comment by Boluwatifey1** [comment]
Hello @srish, I've implemented a new design for the step title and description fields. For step titles, I've introduced a separate field marked with a red asterisk (*) to signify that it's a required field, aligning with best practices for clarity.

Regarding the description field, I've integrated a variety of valuable text customization options, including font size, font color, alignment, strikethrough, underline, italics, numbered tasks, and bulleted tasks. These features empower users to personalize their text and increase the flexibility and expressiveness of their content, ultimately enhancing the overall user experience.

I'd appreciate your feedback on these enhancements and whether you think they align with our goals for user satisfaction. Your insights are invaluable as we continue to refine our platform for a more user-friendly and personalized experience.
Here is the design:
![ACTIVITY CREATION (4)](https://github.com/unstructuredstudio/zubhub/assets/127180289/b9b3226b-e67a-4f7f-b90c-dd47f6476cc9)

**Comment by GloriaKaduru** [comment]
Hi, @srish ! I worked on a few explorations for the UX improvement of the '_Step_' section and here's what I came up with.

**Exploration 1**
Like you suggested, adding an input field for _Step Title_ input with a label and placeholder is a better idea visually as well as for accessibility. I worked on trying to improve the ux of the entire step creation process by adding drowpdown buttons that expand and collapse the various steps added to that particular activity to reduce clutter and prevent long scrolling by shortening the form. I also added lines to show demarcation between the various steps to aid cohesion.
![Step Section Exploration 1](https://github.com/unstructuredstudio/zubhub/assets/82684587/9b13f1ae-c548-4563-a65d-d2ac61a29f51)

**Exploration 2**
This design is pretty similar to the one above, with minor differences in the placeholder copy and in the asterisk added to _Title_  to indicate a required field. _Description_ isn't set as required because the creator may choose to upload images to describe the step instead of using text.
![Step Section Exploration 2](https://github.com/unstructuredstudio/zubhub/assets/82684587/95cd01a2-093d-4be9-99c5-8e2c762a5916)

**Exploration 3**
Here, I made the input fields have floating labels that show after being clicked on to give a more minimalist design consistent with the login/signup forms. I showed what the fields look like while active.

**1. With Title active**
![Step Section Exploration 6](https://github.com/unstructuredstudio/zubhub/assets/82684587/b88b2b9d-2b37-4e6f-b594-7e064720d78d)

**2. With Description active**
![Step Section Exploration 7](https://github.com/unstructuredstudio/zubhub/assets/82684587/be4b94f6-c5af-46cf-a553-f2f0dad2c904)

Please let me know which of these fits better. Thank you!

**Comment by DonPresh** [comment]
> ![Title](https://user-images.githubusercontent.com/105458062/275173276-60b4652c-7018-4cd0-a2a7-9d0e0dc68d3a.png) ![Title](https://user-images.githubusercontent.com/105458062/275173444-2536845b-39f0-4e3c-8140-af4166d2b047.png)
> 
> Hi @srish I was thinking we could add a _required_ text so users can understand better but I couldn't decide which one looks better so here's both.

lol. I literary thought/ did something similar.

**Comment by DonPresh** [comment]
### First iteration
Added a text field and an indicator to show steps to be taken.
![Step 1](https://github.com/unstructuredstudio/zubhub/assets/100778015/4a2eacbb-189b-439e-bbec-a0472feef036)

### Second iteration
Added a 'required' text suggestions.
![step](https://github.com/unstructuredstudio/zubhub/assets/100778015/181c94e9-5703-4a10-92c3-020a136fde62)

**Comment by Adedolapooo** [comment]
Hello @srish. I noticed that this screenshot was for step one , so I automatically thought there must be more steps to follow. I then proceeded to add a progress bar so that the user can follow through the creation of the form. I also redesigned the input fields to help the user achieve their task. See below; 

![94A921B5-7B3B-46E1-903A-87871CC23E4B](https://github.com/unstructuredstudio/zubhub/assets/105403711/8075727f-23e0-4a9f-988e-57e11f4cf4e4)

I’m open to corrections and suggestions. Thank you
